Gems/Atom/RPI/Code/Source/RPI.Public/Pass/AttachmentReadback.cpp

```diff
@@ -339,7 +355,10 @@ namespace AZ
                     }
                     if (m_callback)
                     {
-                        m_callback(GetReadbackResult());
+                        for (const auto& readbackItem : m_readbackItems)
+                        {
+                            m_callback(GetReadbackResult(readbackItem));
+                        }
```

Comment:
It would be better if the m_callback is only called once for all the mips. 
Although this requires change with ReadbackResult structure to include all the mips' data.

Reply:
I'm going to modify `struct ReadbackItem` with new member variables (the new list of data buffers), so the existing code base does not get impacted. And keep it as a one-time dispatched callback.

Reply:
Done. Now the callback is executed once.